### PR TITLE
Extract wordpress externals 31351

### LINKED ISSUE #31351: Refactor build tool: Extract `@wordpress/` externals
https://github.com/Automattic/wp-calypso/blob/40f010e7f0d891e69464ab849822e18b4fd707ca/webpack.config.js#L134-L158

### PR DESCRIPTION
**Pull request by aldavigdis**
#### Changes proposed in this Pull Request

- Moving most @WordPress external dependencies to the calypso-build package.
- babel-related dependencies stay in the main package.json file as they seem too interwoven and splitting them away requires more work than expected, so it requires a separate issue.

#### Testing instructions

1. Delete your local `./node_modules/` directory to uninstall all local dependenceis
2. Run `npm install` and then `npm start` and see if anything breaks.
3. Try building the blocks

Closes #31351

🍐 with @gititon

### CODE REVIEW

File: packages/calypso-build/webpack/wordpress-externals.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+const wordpressRequire = request => {
+	// @wordpress/components -> [ @wordpress, components ]
+	const [ , name ] = request.split( '/' );
+
+	// components -> wp.components
+	return `wp.${ name.replace( /-([a-z])/g, ( match, letter ) => letter.toUpperCase() ) }`;
+};
+
+const wordpressExternals = ( context, request, callback ) =>
```
Comment (enejb) [inline packages/calypso-build/webpack/wordpress-externals.js]: Why not just `module.exports.wordpressExternals` instead?

**Comment by aldavigdis** [comment]
@roccotripaldi seems to have managed to fix the issue. I'm going to squash this.

**Comment by aldavigdis** [comment]
Should we commit `npm-shrinkwrap.json` as well here?

File: package.json [hunk 2]
```diff
@@ -34,33 +34,12 @@
...
-    "@wordpress/hooks": "2.0.5",
-    "@wordpress/i18n": "3.1.1",
-    "@wordpress/keycodes": "2.0.6",
-    "@wordpress/notices": "1.1.3",
-    "@wordpress/nux": "3.0.9",
-    "@wordpress/plugins": "2.0.11",
-    "@wordpress/rich-text": "3.0.7",
-    "@wordpress/token-list": "1.1.0",
-    "@wordpress/url": "2.3.3",
-    "@wordpress/viewport": "2.1.1",
```
Comment (sirreal) [inline package.json]: Why are we changing dependencies in this PR? How did we arrive at this list?

These deps are likely extraneous after #31199 and the removal of Gutenberg-in-Calypso, but the don't seem related to the changes here.

File: package.json [hunk 3]
```diff
@@ -34,33 +34,12 @@
     "@babel/preset-env": "7.3.4",
     "@babel/preset-react": "7.0.0",
     "@babel/runtime": "7.3.4",
-    "@wordpress/a11y": "2.0.2",
-    "@wordpress/api-fetch": "2.2.8",
```
Comment (ockham) [inline package.json]: Since a lot of these are used by Jetpack blocks, we need to test those to make sure they're still working (which they should, since their dependencies are in `packages/jetpack-blocks/package.json`. But better safe than sorry :sweat_smile: )

File: package.json (hunk 2 above)
Comment (ockham) [reply package.json]: Fair point, it probably makes sense to tackle this in a different PR.

Comment (roccotripaldi) [reply package.json]: Nice! Good call.

File: package.json (hunk 3 above)
Comment (roccotripaldi) [reply package.json]: I added the deps back. We can handle their removal in an other PR

File: npm-shrinkwrap.json [hunk 4]
```diff
@@ -2165,9 +2165,9 @@
       "integrity": "sha512-shAmDyaQC4H92APFoIaVDHCx5bStIocgvbwQyxPRrbUY20V1EYTbSDchWbuwlMG3V17cprZhA6+78JfB+3DTPw=="
```
Comment (sirreal) [inline npm-shrinkwrap.json]: Let's revert these changes since we're not changing deps.

Comment (roccotripaldi) [reply npm-shrinkwrap.json]: Nice! Good call.
